lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -733,7 +734,13 @@ def set_zsort(self, zsort):
             The function applied on the z-coordinates of the vertices in the
             viewer's coordinate system, to determine the z-order.
         """
-        self._zsortfunc = self._zsort_functions[zsort]
+        def nansafe(func):
+            def f(x):
+                value = func(x)
+                return sys.maxsize if np.isnan(value) else value
```

Comment:
Let me know if there is a better way than using `sys.maxsize`

Reply:
Would `np.inf` be better?

Reply:
Yes you are right, not sure why I didn't do that in the first place. I've changed it now.